pkg/store/etcd.go

```diff
@@ -115,6 +110,30 @@ func (s *Etcd) Put(key string, value []byte) error {
 	return nil
 }
 
+// Create the entire path for a directory that does not exist
+func (s *Etcd) createDirectory(path string) error {
+	if _, err := s.client.CreateDir(normalize(path), 10); err != nil {
+		if etcdError, ok := err.(*etcd.EtcdError); ok {
+			if etcdError.ErrorCode != 105 { // Skip key already exists
+				return err
+			}
+		} else {
+			return err
+		}
+	}
+	return nil
+}
+
+// Periodically renew a key with an Ephemeral flag set
+func (s *Etcd) periodicRenew(key string, value []byte, ttl time.Duration) {
+	renew := time.NewTicker(40 * time.Second)
+	for _ = range renew.C {
+		if _, err := s.client.Set(key, string(value), uint64(ttl)); err != nil {
```

Comment:
This will override the key, meaning that:

```
c.Put("foo", "bar", &WriteOptions{Ephemeral: true})
c.Put("foo", "hello world")
c.Get("foo") <- after 40 seconds this is "bar" again
```

You can solve this by using an `CompareAndSwap` and if it fails, at that point do a get. 
Probably no need to get the `value` as a parameter: just go a get first, grab the lastIndex and value, then compare and swap with that value and last index. When it fails, get the new value and lastIndex and loop on